packages/next-server/server/next-server.ts

```diff
@@ -160,6 +160,29 @@ export default class Server {
     ]
 
     if (this.nextConfig.useFileSystemPublicRoutes) {
+      if (this.nextConfig.experimental.amp) {
+        // It's very important to keep this route's param optional.
+        // (but it should support as many params as needed, separated by '/')
+        // Otherwise this will lead to a pretty simple DOS attack.
+        // See more: https://github.com/zeit/next.js/issues/2617
+        routes.push({
+          match: route('/amp/:path*'),
```

Comment:
I'd recommend going with a route like `:path/amp` or similar, so that automatic URL shortening like 
<img width="237" alt="screen shot 2019-02-06 at 8 38 19 am" src="https://user-images.githubusercontent.com/89679/52357218-97f4ce00-29ea-11e9-9a70-ff1dd14b4bad.png">
is not changed at all.

Reply:
Sounds good to me! Implementing 👍